activerecord/test/cases/adapters/mysql2/sanitize_test.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+require "models/post"
+
+class Mysql2SanitizeTest < ActiveRecord::Mysql2TestCase
+  def setup
+    super
+  end
+
+  def test_sanitize_sql_for_conditions_from_docs
+    actual = ActiveRecord::Base.sanitize_sql_for_conditions(["name=? and group_id=?", "foo'bar", 4])
+    expected = "name='foo\\'bar' and group_id='4'"
+    assert_equal expected, actual
+
+    actual = ActiveRecord::Base.sanitize_sql_for_conditions(["name=:name and group_id=:group_id", name: "foo'bar", group_id: 4])
+    expected = "name='foo\\'bar' and group_id='4'"
+    assert_equal expected, actual
+
+    actual = ActiveRecord::Base.sanitize_sql_for_conditions(["name='%s' and group_id='%s'", "foo'bar", 4])
+    expected = "name='foo\\'bar' and group_id='4'"
+    assert_equal expected, actual
+
+    actual = ActiveRecord::Base.sanitize_sql_for_conditions("name='foo''bar' and group_id='4'")
+    expected = "name='foo''bar' and group_id='4'"
+    assert_equal expected, actual
+  end
+
+  def test_sanitize_sql_for_assignment_from_docs
+    actual = Post.sanitize_sql_for_assignment(["name=? and group_id=?", nil, 4])
+    expected = "name=NULL and group_id='4'"
+    assert_equal expected, actual
+
+    actual = Post.sanitize_sql_for_assignment(["name=:name and group_id=:group_id", name: nil, group_id: 4])
+    expected = "name=NULL and group_id='4'"
+    assert_equal expected, actual
+
+    actual = Post.sanitize_sql_for_assignment({ name: nil, group_id: 4 })
+    expected = "`posts`.`name` = NULL, `posts`.`group_id` = 4"
+    assert_equal expected, actual
+
+    actual = Post.sanitize_sql_for_assignment("name=NULL and group_id='4'")
+    expected = "name=NULL and group_id='4'"
+    assert_equal expected, actual
+  end
+
+  def test_sanitize_sql_for_order_from_docs
+    skip "See https://github.com/rails/rails/issues/44312"
+    actual = Post.sanitize_sql_for_order([Arel.sql("field(id, ?)"), [1,3,2]])
+    expected = "field(id, 1,3,2)"
+    assert_equal expected, actual
+
+    actual = Post.sanitize_sql_for_order("id ASC")
+    expected = "id ASC"
+    assert_equal expected, actual
+  end
+
+  def test_sanitize_sql_hash_for_assignment_from_docs
+    actual = ActiveRecord::Base.sanitize_sql_hash_for_assignment({ status: nil, group_id: 1 }, "posts")
+    expected = "`posts`.`status` = NULL, `posts`.`group_id` = 1"
+    assert_equal expected, actual
+  end
+
+  def test_sanitize_sql_like_from_docs
```

Comment:
Technically the `like` tests do not depend on the connection adapter. But they easily could in future. I always opt for better and proactive test coverage, so added the same tests in all three adapters.